### ISSUE TITLE
Testing on trained data using predict_generator gives weid confusion matrix

### ISSUE THREAD

**Issue by aparajita31pandey** [aymericdamien/TensorFlow-Examples#300]
My model is giving 90% accuracy on train and validation data during training . and after that using predict generator on that same  train data gives very unsatisfying confusion matrix and  givies very nice confusion matrix on validation data. I don't understand why so 
Confusion Matrix on validation data:-
predictions = model.predict_generator(validation_generator,
                                      steps=validation_generator.samples / validation_generator.batch_size, verbose=1)
[[16  4]
 [ 3 17]]

Confusion Matrix On training Data
predictions = model.predict_generator(train_generator,
                                      steps=train_generator.samples / train_generator.batch_size, 
[[ 60 225]
 [232 112]]

Accuracy and Loss graph on both these data seems to look pretty good though.
Any help would be highly appreciated
